src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php

```diff
@@ -26,6 +28,9 @@ class ValidationListener implements EventSubscriberInterface
     private ValidatorInterface $validator;
     private ViolationMapperInterface $violationMapper;
 
+    /** @var FormInterface[][] */
+    private static array $dispatchEvents = [];
```

Comment:
Can you please provide more details? I dont see how this could help.

`ValidationListener` is dispatched on all Form items, but only the root form will trigger the validation by recursively validate all children. (all other form items are just ignored)

Now, instead of just ignoring them, I use a static property to collect form items that are listening to the new events. Because we cannot dispatch them immediately, we need to wait to be inside the `ValidationListener` of the root form which is the only one that will really start the form validation.

Reply:
The suggestion is about removing the `static` condition of this property which makes the `ValidationListener` class stateful. Unless there is no other way to achieve it we should avoid this practice.

In `FormTypeValidatorExtension` you could do the following:
```php
class FormTypeValidatorExtension extends BaseValidatorExtension
{
    private ValidationListener $validationListener;

    public function __construct(ValidatorInterface $validator, bool $legacyErrorMessages = true, FormRendererInterface $formRenderer = null, TranslatorInterface $translator = null)
    {
        $this->validationListener = new ValidationListener($validator, new ViolationMapper($formRenderer, $translator));
    }

    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $builder->addEventSubscriber($this->validationListener);
    }

   // ...
}
```
passing the same instance to every form (improving performance a bit) and then collecting all forms (non-static property) without affecting the state of the class.

Reply:
I see. I thought every Form item were using his own instance of `FormTypeValidatorExtension`, but this is not the case.
So your suggestion works. 👍🏻 
I pushed the changes.